apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,379 @@
+the type of menu items in the list can vary.
+
+```tsx
+<Menu>
+  <MenuTrigger>
+    <Button>Toggle menu</Button>
+  </MenuTrigger>
+  <MenuPopover>
+    <MenuList>
+      <MenuGroup>
+        <MenuGroupHeader>Create New Items</MenuGroupHeader>
+        <MenuItem>New Project</MenuItem>
+        <MenuItem>New Timeline</MenuItem>
+      </MenuGroup>
+      <MenuDivider />
+      <MenuItem disabled>Open File...</MenuItem>
+      <MenuItem>Open Folder...</MenuItem>
+      <MenuDivider />
+      <Menu>
+        <MenuTrigger>
+          <MenuItem>Zoom</MenuItem>
+        </MenuTrigger>
+        <MenuPopover>
+          <MenuList>
+            <MenuItem>Zoom In</MenuItem>
+            <MenuItem>Zoom Out</MenuItem>
+          </MenuList>
+        </MenuPopover>
+      </Menu>
+    </MenuList>
+  </MenuPopover>
+</Menu>
+```
+
+## Usage examples
+
+### Passing content
+
+You can pass a primitive value to slots.
+The `Input` component's `contentBefore` and `contentAfter` slots can be passed strings.
+
+```tsx
+<Input contentBefore="$" value="10" contentAfter=".00" />
+```
+
+You can pass HTML values to slots.
+The `Button` component can be passed an `img`.
+
+```tsx
+<Button icon={<img src='site-icon.png' alt='branded site icon' /> }
+```
+
+You can pass JSX/TSX to slots.
+The `Button` component can be passed an `CalendarRegular24` icon.
+
+```tsx
+<Button icon={<CalendarRegular24 />}
+```
+
+### Passing props
+
+When a slot renders a component or element by default, you can pass props to it.
+
+The `Avatar` component has a `badge` slot that renders a `PresenceBadge`.
+This allows you pass `PresenceBadge` props.
+
+```tsx
+<Avatar name="Support" badge={{ status: 'available', 'aria-label': 'available' }} />
+```
+
+You can pass the `className` prop to customize the style of a slot.
+The `Avatar` badge slot supports a `className`, allowing the badge style to be customized.
+
+```tsx
+const useStyles = makeStyles({
+  badge: { color: tokens.colorBrandStroke1 },
+});
+
+const BusyBrandAvatar = () => {
+  const styles = useStyles();
+  return <Avatar name="IT probably" badge={{ status: 'busy', className: styles.badge }} />;
+};
+```
+
+You can pass the `as` prop to change the element type of a slot.
+Note that you must choose from one of the available element types the slot supports.
+
+```tsx
+<AccordionHeader as="h1">Accordion Header as h1</AccordionHeader>
+```
+
+### Replacing the entire slot
+
+When you pass content or props to a slot,
+the component renders the content within a component or element based on the slot type.
+If you need to replace the slot's entire content, including the containing element,
+pass a render function as the children.
+
+This is an escape hatch in the slots API, so prefer the other techniques whenever possible.
+
+By passing `renderBigLetterIcon` as the `children`, the `span` that normally contains the icon is replaced with an `h1`.
+
+```tsx
+const renderBigLetterIcon () => {
```

Comment:
```suggestion
const renderBigLetterIcon = (Component, props) => {
```

I think that's is reasonable to show input there so people will be aware about it.